neat_cache/lib/src/providers/resp.dart

```diff
@@ -0,0 +1,514 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// Minimalistic [RESP[1] protocol implementation in Dart.
+///
+/// [1]: https://redis.io/topics/protocol
+library resp;
+
+import 'dart:async';
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:logging/logging.dart';
+
+final _log = Logger('neat_cache:redis');
+
+/// Thrown when a command returns
+class RedisErrorException implements Exception {
+  final String type;
+  final String message;
+
+  RedisErrorException._(this.type, this.message);
+
+  @override
+  String toString() => 'RedisErrorException: $type $message';
+}
+
+/// Thrown if the redis connection is broken.
+///
+/// This typically happens if the connection is unexpectedly closed, the
+/// [RESP protocol][1] is violated, or there is an internal error.
+///
+/// [1]: https://redis.io/topics/protocol
+class RedisConnectionException implements Exception {
```

Comment:
Could this be an `IOException`? or a `FormatException`?

Reply:
Are you suggesting reuse those instead, or subclassing them?

I found that there is a lot of exception that a consumer of `resp.dart` doesn't need to know about... The essence is the same: connection has stopped working and you need to reconnect. That's what this exception is.

Reply:
I was thinking `implements IOException`. I think it is fine like this, but for database-like clients I like to have the separation of the errors being IO and non-IO...

Reply:
I just have mixed feelings about dependency on `dart:io`.